core/ir/aarch64/codec.txt

```diff
@@ -1057,12 +1068,38 @@ x101101011000000000101xxxxxxxxxx  cls     wx0 : wx5
 0x001110xx1xxxxx100001xxxxxxxxxx     add       dq0 : dq5 dq16 bhsd_sz
 0x001110xx1xxxxx100011xxxxxxxxxx     cmtst     dq0 : dq5 dq16 bhsd_sz
 0x001110xx1xxxxx100101xxxxxxxxxx     mla       dq0 : dq0 dq5 dq16 bhs_sz
-0x101111xxxxxxxx0000x0xxxxxxxxxx     mla       dq0 : dq5 dq16_h_sz bhsd_sz index_lhm
+0x101111xxxxxxxx0000x0xxxxxxxxxx     mla       dq0 : dq5 imm4_16 bhsd_sz index_lhm
```

Comment:
AIUI `imm4_16` is a combination of element size, `M` bit and source register encoding and is used in combination with `index_lhm`. I think it needs to have a name which reflects this behaviour, e.g. `dq16_idx_lhm`? Also worth mentioning this mutual interdependence in the comments for both fields' decode functions in `codec.c `as well as the comments for the bitfields' definitions above.

Reply:
imm4_16 is literally just encoding/decoding bits 20:16 (not M:20:16 like previously) as an immediate. I did this purposefully to seperate it from index_lhm to avoid the limitations that the codec faces with interdependent operands

Reply:
done, changed `imm4_16` to `dq16_idx_lhm` and `index_lhm` to `idx_lhm`